Funcky.Analyzer/Funcky.Analyzer.CodeFixes/EnumerableRepeatNeverCodeFix.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Funcky.Analyzer.CodeFixResources;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Funcky.Analyzer
+{
+    [Shared]
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(EnumerableRepeatNeverCodeFix))]
+    public sealed class EnumerableRepeatNeverCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(EnumerableRepeatNeverAnalyzer.DiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider()
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnosticSpan = context.Diagnostics.First().Location.SourceSpan;
+            var declaration = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First();
+
+            context.RegisterCodeFix(CreateFix(context, declaration), GetDiagnostic(context));
+        }
+
+        private static Diagnostic GetDiagnostic(CodeFixContext context)
+            => context.Diagnostics.First();
+
+        private CodeAction CreateFix(CodeFixContext context, InvocationExpressionSyntax declaration)
+            => CodeAction.Create(
+                EnumerableRepeatNeverCodeFixTitle,
+                CreateSequenceReturnAsync(context.Document, declaration),
+                nameof(EnumerableRepeatNeverCodeFixTitle));
+
+        private Func<CancellationToken, Task<Document>> CreateSequenceReturnAsync(Document document, InvocationExpressionSyntax declaration)
+            => async cancellationToken
+                => document.WithSyntaxRoot(await ReplaceWithSequenceReturn(document, declaration, cancellationToken).ConfigureAwait(false));
+
+        private async Task<SyntaxNode> ReplaceWithSequenceReturn(Document document, InvocationExpressionSyntax declaration, CancellationToken cancellationToken)
+        {
+            var oldRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync();
+
+            return oldRoot.ReplaceNode(declaration, CreateEnumerableReturnRoot(ExtractFirstArgument(declaration), semanticModel));
+        }
+
+        private static ArgumentSyntax ExtractFirstArgument(InvocationExpressionSyntax invocationExpr)
+            => invocationExpr.ArgumentList.Arguments[Argument.First];
+
+        private SyntaxNode CreateEnumerableReturnRoot(ArgumentSyntax firstArgument, SemanticModel model)
+            => InvocationExpression(
+                    MemberAccessExpression(
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        IdentifierName(nameof(Enumerable)),
```

Comment:
I wouldn't hardcode these like this since it won't work if `System.Linq` is not imported or if there's a `using Foo = System.Linq.Enumerable;`

Instead you could get the type symbol from the compilation (`ISemanticModel.Compilation`) using `GetTypeByMetadataName` and then use the `ParseTypeName(ToMinimalDisplayString(...))` trick to get a context-aware type name.

Reply:
Verified that this is a Problem.

Reply:
Similar problem is in EnumerableRepeatOnceCodeFix too.

Reply:
I think I got it working, fixed in 59511b88f7efcf27ff5767e3f765936d83f7c856